lib/platform-stats.js

```diff
@@ -11,38 +11,32 @@ const debug = createDebug('spark:platform-stats')
  * @param {import('./preprocess.js').Measurement[]} allMeasurements
  */
 export const updatePlatformStats = async (pgClient, honestMeasurements, allMeasurements) => {
-  const participants = new Set(allMeasurements.map(m => m.participantAddress))
-  await updateDailyParticipants(pgClient, participants)
+  const participantsMap = await mapParticipantsToIds(pgClient, new Set(allMeasurements.map(m => m.participantAddress)))
+  await updateDailyParticipants(pgClient, Array.from(participantsMap.values()))
   await updateDailyStationStats(pgClient, honestMeasurements, allMeasurements)
+  await updateStationsAndParticipants(pgClient, honestMeasurements, allMeasurements, participantsMap)
 }
 
 /**
  * @param {import('pg').Client} pgClient
  * @param {import('./preprocess.js').Measurement[]} honestMeasurements
  * @param {import('./preprocess.js').Measurement[]} allMeasurements
- * @param {Date} day
+ * @param {object} options
+ * @param {Date} [options.day]
  */
 export const updateDailyStationStats = async (
   pgClient,
   honestMeasurements,
   allMeasurements,
-  day = new Date()
+  { day = new Date() } = {}
 ) => {
   /** @type{Map<string, {accepted: number, total: number}>} */
   const statsPerStation = new Map()
 
-  // JSON stringify the key to make it comparable, since objects compare by reference
-  /** @type {(m: import('./preprocess.js').Measurement) => string} */
-  const getKey = (m) => JSON.stringify({
-    stationId: m.stationId,
-    participantAddress: m.participantAddress,
-    inet_group: m.inet_group
-  })
-
   for (const m of honestMeasurements) {
     if (m.stationId == null) continue
 
-    const key = getKey(m)
+    const key = getStatsKey(m)
```

Comment:
> When getKey is defined inside updateDailyStationStats, then its purpose is clear - it calculates the key we use to group measurements for calculating "daily station stats".
>
> When you move it outside of updateDailyStationStats, it's no longer clear what key it computes.

Especially since this will be removed anyway, I don't want to bikeshed about this part. I'm thinking:

- Moving the new function close to this function helps with discoverability. I also don't like when the new function is so far away
- Reverting that part and inline also works, to keep this diff small

Please whoever feels strongly just make a decision and let's finish up this work

Reply:
Will stick with inline, since the change is now outside the scope of this PR